### PR TITLE
Math Evaluators

### PR DESCRIPTION
**Pull request by ninghu**
# Description

Please add an informative description that covers that changes made by the pull request and link all relevant issues.

# All Promptflow Contribution checklist:
- [ ] **The pull request does not introduce [breaking changes].**
- [ ] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
- [ ] **I have read the [contribution guidelines](https://github.com/microsoft/promptflow/blob/main/CONTRIBUTING.md).**
- [ ] **I confirm that all new dependencies are compatible with the MIT license.**
- [ ] **Create an issue and link to the pull request to get dedicated review from promptflow team. Learn more: [suggested workflow](../CONTRIBUTING.md#suggested-workflow).**

## General Guidelines and Best Practices
- [ ] Title of the pull request is clear and informative.
- [ ] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### Testing Guidelines
- [ ] Pull request includes test coverage for the included changes.

### CODE REVIEW

File: src/promptflow-evals/promptflow/evals/_common/utils.py [hunk 1]
```diff
@@ -5,8 +5,20 @@
...
+import nltk
 import numpy as np
 
+try:
+    from nltk.tokenize.nist import NISTTokenizer
+except LookupError:
+    nltk.download("perluniprops")
+    nltk.download("punkt")
+    nltk.download("punkt_tab")
+    from nltk.tokenize.nist import NISTTokenizer
```
Comment (MilesHolland) [inline src/promptflow-evals/promptflow/evals/_common/utils.py:20]: This feels like hidden dependency logic. Why do we need to download stuff to make this import work, and why can't this be covered by the package's dependency list?

**Review by luigiw — APPROVED** [review]

File: src/promptflow-evals/promptflow/evals/_common/utils.py (hunk 1 above)
Comment (ninghu) [reply src/promptflow-evals/promptflow/evals/_common/utils.py:20]: I had the same question ealier. Those are nltk data packages, not standard python packages, so we really cannot just add them to the pyproject.toml dependency list. the only to download them is through the python script.

File: src/promptflow-evals/promptflow/evals/_common/utils.py [hunk 2]
```diff
@@ -28,3 +40,19 @@ def get_harm_severity_level(harm_score: int) -> str:
         if harm_score_range[0] <= harm_score <= harm_score_range[1]:
             return harm_level.value
     return np.nan
+
+
+def nltk_tokenize(text: str) -> List[str]:
```
Comment (singankit) [inline src/promptflow-evals/promptflow/evals/_common/utils.py:45]: Just curious what is this for ?

File: src/promptflow-evals/promptflow/evals/evaluators/_meteor/_meteor.py [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+    **Output format**
+
+    .. code-block:: python
+
+        {
+            "meteor_score": 0.62
+        }
+    """
+
+    def __init__(self, alpha: float = 0.9, beta: float = 3.0, gamma: float = 0.5):
```
Comment (singankit) [inline src/promptflow-evals/promptflow/evals/evaluators/_meteor/_meteor.py:72]: We should add a test to validate if default values gets overwritten.

**Review by singankit — APPROVED** [review]

File: src/promptflow-evals/promptflow/evals/_common/utils.py (hunk 2 above)
Comment (ninghu) [reply src/promptflow-evals/promptflow/evals/_common/utils.py:45]: It's for splitting the sentense into tokens. I have a unit-test which can demosntrate what it does.

File: src/promptflow-evals/promptflow/evals/evaluators/_meteor/_meteor.py (hunk 3 above)
Comment (ninghu) [reply src/promptflow-evals/promptflow/evals/evaluators/_meteor/_meteor.py:72]: Will do it in a follow up.

**Pull request merged**